### DISCUSSION TITLE
Containers in containers

### DISCUSSION THREAD

**Discussion by vedeja** [structurizr/java#259] [Ideas]
We have a large model. I love the concept of the firm C:s, but in reality we would benefit from being able to have containers in containers or maybe even configurable, custom levels. Right now we are using groups to help the fact that we have so many containers but it only goes so long. It feels like I’m maybe cursing here, but what is the likelihood of you relaxing this and allowing more levels?

**Comment by simonbrowndotje** [comment]
> what is the likelihood of you relaxing this and allowing more levels?

Zero likelihood I'm afraid, as it requires a rewrite of almost everything. Adding levels to the C4 model is unnecessary for 95% of teams though. For those that need it, I'd suggest looking at something like Ilograph or LikeC4 (which I see you've discovered).

**Reply by vedeja** [reply]
It makes sense and I was expecting that. It makes me question if and why our team should belong to that other 5%.

**Reply by simonbrowndotje** [reply]
In my experience, and I mention this in my workshops/talks, the need for additional levels of abstraction tends to suggest a misuse of the C4 model and/or the tooling. I don't know what you're modelling, so I am not saying that's the case here.
